src/busctl/busctl.c

```diff
@@ -2281,6 +2281,114 @@ static int get_property(int argc, char **argv, void *userdata) {
         return 0;
 }
 
+static void sd_event_exit_success(sd_event **e) {
+        if (*e)
+                sd_event_exit(*e, 0);
+}
+
+static int on_bus_signal(sd_bus_message *msg, void *userdata, sd_bus_error *ret_error) {
+        _cleanup_(sd_event_exit_success) sd_event *e = NULL;
```

Comment:
Interesting, but any failures should be propagated, I think.
Hence, split out the main implementation of on_bus_signal(), then pass the error code to sd_event_exit().
```c
static int on_bus_signal_impl(sd_bus_message *msg) {
        int r;

        assert(msg);

        r = sd_bus_message_is_empty(...);
        if (r < 0)
                return bus_log_parse_error(r);

        ...
}

static int on_bus_signal(sd_bus_message *msg, void *userdata, sd_bus_error *ret_error) {
        sd_event *e = sd_bus_get_event(sd_bus_message_get_bus(ASSERT_PTR(msg)));
        return sd_event_exit(e, on_bus_signal_impl(msg));
}
```